app/src/main/java/com/github/h3lp3rs/h3lp/professional/VerificationActivity.kt

```diff
@@ -0,0 +1,140 @@
+
+    }
+
+    // Layout components
+    private lateinit var chooseImgButton: Button
+    private lateinit var uploadButton: Button
+    private lateinit var fileName: EditText
+    private lateinit var img: ImageView
+    private lateinit var progressBar: ProgressBar
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_verification)
+
+        chooseImgButton = findViewById(R.id.button_choose_img)
+        uploadButton = findViewById(R.id.button_upload)
+        fileName = findViewById(R.id.edit_file_name)
+        img = findViewById(R.id.file)
+        progressBar = findViewById(R.id.progress_bar)
+
+        chooseImgButton.setOnClickListener {
+            openFileChooser()
+        }
+        uploadButton.setOnClickListener {
+            uploadFile()
+        }
+    }
+
+    /**
+     * Opens the image file chooser of the device inorder to select an image
+     */
+    private fun openFileChooser() {
+        val intent = Intent()
+        intent.type = INTENT_TYPE
+        intent.action = Intent.ACTION_GET_CONTENT
+        resultLauncher.launch(intent)
+    }
+
+    /**
+     * Handles the intent result launched by the image file selection
+     */
+    private val resultLauncher =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode == Activity.RESULT_OK && result.data != null && result.data!!.data != null) {
+                imgUri = result.data!!.data!!
+                // Display the selected image
+                Picasso.with(this).load(imgUri).into(img)
+            }
+        }
+
+    /**
+     * Retrieves the file extension
+     *
+     * @param uri Uri of the file
+     * @return the extension of the file
+     */
+    private fun getFileExtension(uri: Uri): String? {
+        return MimeTypeMap.getSingleton().getExtensionFromMimeType(contentResolver.getType(uri))
+    }
+
+    /**
+     * Uploads the selected file into the cloud storage as well as the stores the current user in the authenticated users
+     * database
+     */
+    private fun uploadFile() {
+        if (imgUri != null) {
+            // Creates a file in the cloud storage with the current time as name (Sufficient to avoid collision but can be redesigned in the future )
+            val fileReference: StorageReference = storageRef.child(
+                System.currentTimeMillis().toString() + "." + getFileExtension(
+                    imgUri!!
+                )
+            )
+            // Stores the file the cloud storage
+            fileReference.putFile(imgUri!!).addOnSuccessListener {
+                // A delay to be able to see the progress bar at 100% before redirection
+                Handler(Looper.getMainLooper()).postDelayed({
+                    progressBar.progress = 0
+                }, 5000)
```

Comment:
Really like the fact that the animation will terminate every time, you could maybe put the 5000 in some sort of constant?